georocket-server/src/main/java/io/georocket/GeoRocket.java

```diff
@@ -552,52 +552,54 @@ protected static String getGeoRocketHomeStr() {
    * Get the GeoRocket home as {@link File}
    * @param geoRocketHomeStr the path to the GeoRocket home
    * @return the GeoRocket home file
+   * @throws IOException if
    */
-  protected static File getRocketHomeFile(String geoRocketHomeStr) {
-    try {
-      return geoRocketHome = new File(geoRocketHomeStr).getCanonicalFile();
-    } catch (IOException e) {
-      log.error("Invalid GeoRocket home: " + geoRocketHomeStr);
-      System.exit(1);
-      return null; // Will never happen
-    }
+  protected static File toRocketHomeFile(String geoRocketHomeStr) throws IOException {
+    return new File(geoRocketHomeStr).getCanonicalFile();
   }
 
   /**
    * Load the GeoRocket configuration and return it.
    * Will return an empty configuration if something went wrong.
    * @param geoRocketHome the GeoRocket home path
    * @return the configuration
+   *
+   * @throws IllegalStateException If the configuration file could not be found
+   * @throws DecodeException If failed to decode the JSON configuration
    */
-  protected static JsonObject loadConfiguration(File geoRocketHome) {
+  protected static JsonObject loadConfiguration(File geoRocketHome) throws IOException, DecodeException {
     File confDir = new File(geoRocketHome, "conf");
     File confFile = new File(confDir, "georocketd.json");
 
     try {
       String confFileStr = FileUtils.readFileToString(confFile, "UTF-8");
       return new JsonObject(confFileStr);
     } catch (IOException e) {
```

Comment:
Catching an IOException and then wrapping it in another one doesn't make much sense.